app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddSmoothnessForm.kt

```diff
@@ -0,0 +1,166 @@
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.AImageListQuestAnswerFragment
+import de.westnordost.streetcomplete.quests.OtherAnswer
+import de.westnordost.streetcomplete.quests.surface.Surface
+import de.westnordost.streetcomplete.quests.surface.asItem
+import de.westnordost.streetcomplete.view.image_select.Item
+import de.westnordost.streetcomplete.view.image_select.ItemViewHolder
+
+class AddSmoothnessForm : AImageListQuestAnswerFragment<Smoothness, SmoothnessAnswer>() {
+
+    override val otherAnswers = listOf(
+        OtherAnswer(R.string.quest_smoothness_wrong_surface) { surfaceWrong() },
+        OtherAnswer(R.string.quest_smoothness_obstacle) { showObstacleHint() }
+    )
+
+    private val surfaceTag get() = osmElement!!.tags["surface"]
+
+    private val highwayTag get() = osmElement!!.tags["highway"]
+
+    // show only answers that make sense for the tagged surface
+    // numbers 0-7 are excellent - impassible, ordered like Smoothness class
+    // TODO: leave unused surfaces in here for future extension of the quest? or remove?
+    override val items get() = when {
+        // paving stones on roads will not be "excellent" as they get easily damaged by cars
+        surfaceTag == "paving_stones" && !ALL_PATHS_EXCEPT_STEPS.contains(highwayTag) -> getAnswers(1,4)
+        listOf("asphalt", "paving_stones", "concrete", "metal").contains(surfaceTag) -> getAnswers(0,4)
+        listOf("sett", "unhewn_cobblestone", "compacted", "grass_paver").contains(surfaceTag) -> getAnswers(2,6)
+        listOf("gravel", "fine_gravel", "pebbles").contains(surfaceTag) -> getAnswers(3,6)
+        listOf("dirt", "grass").contains(surfaceTag) -> getAnswers(3,7)
+        else -> getAnswers(0,7)
+    }
+
+    override val itemsPerRow = 1
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        imageSelector.cellLayoutId = R.layout.cell_labeled_icon_select_smoothness
+    }
+
+    override val moveFavoritesToFront = false
+
+    override fun onClickOk(selectedItems: List<Smoothness>) {
+        applyAnswer(SmoothnessValueAnswer(selectedItems.single().osmValue))
+    }
+
+    private fun showObstacleHint() {
+        activity?.let { AlertDialog.Builder(it)
+            .setMessage(R.string.quest_smoothness_obstacle_hint)
+            .setPositiveButton(android.R.string.ok, null)
+            .show()
+        }
+    }
+
+    // this requires Smoothness class to be sorted from best to worst!
+    private fun getAnswers(minSmoothness: Int, maxSmoothness: Int): List<Item<Smoothness>> {
+        val answers = mutableListOf<Item<Smoothness>>()
+        for (smoothness in Smoothness.values().sliceArray(minSmoothness..maxSmoothness))
+            answers.add(Item(smoothness, getImage(smoothness), smoothness.title, getDescription(smoothness)))
+        return answers
+    }
+
+    private fun getDescription(smoothness: Smoothness) = when (smoothness) {
```

Comment:
getDescription and getImage (and to be implemented getTitle) should move to the to be implemented SmoothnessItem.kt.

Also, I think it would be more convenient and more in line with other code if the function was an extension function on `Smoothness`. I.e.

```kotlin
fun Smoothness.getDescriptionResId(surface: string) = when(this) {
  ...
}
```

Reply:
done